### PR TITLE
[ECP-8637] - Migrate to Boleto web component

### PR DESCRIPTION
**Pull request by khushboo-singhvi**
**Description**
Instead of building the forms on the frontend manually, using component for Boleto.
Updating the Boleto Payment Method as per the component and removing unuse block of code

**Tested scenarios**
Successful test for Boleto was completed

Fixes

### CODE REVIEW

File: Gateway/Request/CheckoutDataBuilder.php [hunk 1]
```diff
@@ -168,17 +168,17 @@ public function build(array $buildSubject): array
...
-
-            if (count($boletoTypes) == 1) {
-                $requestBody['selectedBrand'] = $boletoTypes[0];
-                $requestBodyPaymentMethod['type'] = $boletoTypes[0];
-            } else {
-                $requestBody['selectedBrand'] = $payment->getAdditionalInformation("boleto_type");
-                $requestBodyPaymentMethod['type'] = $payment->getAdditionalInformation("boleto_type");
-            }
+        if ($payment->getMethod() == 'adyen_boleto') {
+//            $boletoTypes = $this->configHelper->getAdyenBoletoConfigData('boletotypes');
```
Comment (candemiralp) [inline Gateway/Request/CheckoutDataBuilder.php:172]: Shall we remove commented out part?

File: etc/adminhtml/system/adyen_alternative_payment_methods.xml [hunk 2]
```diff
@@ -46,16 +46,5 @@
             <tooltip>This will override the default shopper locale of the Magento store (e.g. nl_NL). Leave empty to let Magento decide.</tooltip>
             <config_path>payment/adyen_hpp/shopper_locale</config_path>
         </field>
-        <field id="cctypes" translate="label" type="multiselect" sortOrder="60" showInDefault="1" showInWebsite="1" showInStore="0">
-            <label>Boleto Types</label>
-            <source_model>Adyen\Payment\Model\Config\Source\BoletoType</source_model>
-            <config_path>payment/adyen_boleto/boletotypes</config_path>
-        </field>
-        <field id="delivery_days" translate="label" type="text" sortOrder="70" showInDefault="1" showInWebsite="1" showInStore="0">
```
Comment (candemiralp) [inline etc/adminhtml/system/adyen_alternative_payment_methods.xml:54]: Don't we need `delivery_days`? Did you double check with the product?

File: view/adminhtml/templates/info/adyen_boleto.phtml [hunk 3]
```diff
@@ -44,20 +44,6 @@ $_isDemoMode = $block->isDemoMode();
...
-    </div>
-<?php endif; ?>
-
-<?php if ($_info->getAdditionalInformation('boleto_type') != ""): ?>
-    <div>
-        <?= $block->escapeHtml(__('Boleto Type: %1', $_info->getAdditionalInformation('boleto_type'))); ?>
-    </div>
-<?php endif; ?>
-
 <?php if ($_info->getAdditionalInformation('firstname') != ""): ?>
```
Comment (candemiralp) [inline view/adminhtml/templates/info/adyen_boleto.phtml:47]: Do we have `firstname` and `lastname` values in payment additional information after component implementation?

File: view/frontend/web/js/view/payment/adyen-methods.js [hunk 4]
```diff
@@ -43,7 +42,7 @@ define(
                 component: customMethodRenderers[index]
             });
         });
-
+debugger;
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/adyen-methods.js:45]: Debugger needs to be removed.

File: view/frontend/web/js/view/payment/method-renderer/adyen-boleto-method.js [hunk 5]
```diff
@@ -9,117 +9,38 @@
...
-                return this;
-            },
-            validate: function () {
-                let form = 'form[data-role=adyen-boleto-form]';
 
-                let validate = $(form).validation() && $(form).validation('isValid');
+            buildComponentConfiguration: function (paymentMethod, paymentMethodsExtraInfo) {
+                let baseComponentConfiguration = this._super();
+                let formattedBillingAddress = {};
+debugger;
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-boleto-method.js:27]: Debugger needs to be removed.

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js [hunk 6]
```diff
@@ -81,7 +81,7 @@ define(
             createCheckoutComponent: async function(paymentMethodsResponse) {
                 // Set to null by default and modify depending on the paymentMethods response
                 this.adyenPaymentMethod(null);
-
+                console.log("In debug mode");
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:84]: Debugger needs to be removed.

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js [hunk 7]
```diff
@@ -272,6 +271,10 @@ define(
                         'method': this.item.method
                     };
 
+                    debugger
+                    console.log(this.paymentMethod().type);
```
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:275]: Debugger needs to be removed.

File: view/adminhtml/templates/info/adyen_boleto.phtml (hunk 3 above)
Comment (khushboo-singhvi) [reply view/adminhtml/templates/info/adyen_boleto.phtml:47]: Removed

File: etc/adminhtml/system/adyen_alternative_payment_methods.xml (hunk 2 above)
Comment (khushboo-singhvi) [reply etc/adminhtml/system/adyen_alternative_payment_methods.xml:54]: After discussion, we are retaining this field

File: etc/adminhtml/system/adyen_alternative_payment_methods.xml [hunk 8]
```diff
@@ -46,13 +46,7 @@
...
-            <label>Boleto Days for Delivery</label>
```
Comment (candemiralp) [inline etc/adminhtml/system/adyen_alternative_payment_methods.xml:55]: Label is missing for the field.

File: view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-paypal-method.js:1]: This file is irrelevant for this PR. Could you please revert the changes?

File: view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js
Comment (candemiralp) [inline view/frontend/web/js/view/payment/method-renderer/adyen-pm-method.js:1]: This file is irrelevant for this PR. Could you please revert the changes?

**Review by candemiralp — APPROVED** [review]

File: Gateway/Response/CheckoutPaymentsDetailsHandler.php [hunk 9]
```diff
@@ -44,7 +43,7 @@ public function handle(array $handlingSubject, array $response)
 
         // Email sending is set at CheckoutDataBuilder for Boleto
         // Otherwise, we don't want to send a confirmation email
-        if ($payment->getMethod() != AdyenBoletoConfigProvider::CODE) {
+        if ($payment->getMethod() != 'adyen_boleto') {
```
Comment (hossam-adyen) [inline Gateway/Response/CheckoutPaymentsDetailsHandler.php:46]: better to have a constant to avoid any typos

**Review by RokPopov — APPROVED** [review]

**Review by hossam-adyen — DISMISSED** [review]

**Review by hossam-adyen — APPROVED** [review]

**Review by candemiralp — APPROVED** [review]
